<a href="https://colab.research.google.com/github/venkat-vipul/flyrank_ml/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/venkat-vipul/flyrank_ml/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## My Lane

I chose **Refresh / Content Opportunity Scoring** because it focuses on identifying which content pages should be reviewed first using observable search and content signals. This builds directly on the starter notebooks, where I explored relationships in the data and compared rule-based and machine learning approaches for ranking pages. My goal is to develop a decision-support system that helps prioritize limited review effort by producing an evidence-based ranked list of pages for human review, rather than replacing human judgement.

In [ ]:
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Dataset loaded successfully.")
print(f"Rows: {len(df):,}")
print(f"Columns: {len(df.columns)}")

Dataset loaded successfully.
Rows: 30,000
Columns: 44


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

## The Question

**Research Question**

Which content pages should be prioritized for review based on observable search and content signals?

**Decision**

This project aims to improve the decision of which content pages should be reviewed first when review time and resources are limited.

**Action**

A content reviewer or SEO analyst can use the ranked recommendations to decide whether a page should be refreshed, expanded, rewritten, monitored, or left unchanged.

**Cost of a Wrong Recommendation**

A false positive may waste time by directing reviewers to pages that do not require attention. A false negative may leave an important declining page unnoticed, potentially resulting in missed opportunities to improve search performance. The goal is to support better prioritization rather than automate content decisions.

In [ ]:
review_candidates = df[df["trend_direction"] == "down"]

print(f"Potential review candidates: {len(review_candidates):,}")
print(f"Percentage of dataset: {len(review_candidates)/len(df)*100:.1f}%")

Potential review candidates: 16,262
Percentage of dataset: 54.2%


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [ ]:
print(f"Total content pages: {len(df):,}")
print(f"Pages trending down: {(df['trend_direction'] == 'down').sum():,}")
print(f"Median 90-day impressions: {df['impressions_90d'].median():.0f}")
print(f"Median content age (days): {df['content_age_days'].median():.0f}")

Total content pages: 30,000
Pages trending down: 16,262
Median 90-day impressions: 731
Median content age (days): 236


## Quick Look at the Data

The starter dataset contains **30,000** anonymized content pages, providing a substantial set of pages for analysis. Among these, **16,262 pages** are labeled as trending down, suggesting there are many potential candidates for review. The median page has **731 impressions** over the last 90 days and a median content age of **236 days**, indicating that many pages have meaningful search visibility and mature content. These observations suggest that prioritizing which pages should be reviewed first is a worthwhile decision-support problem for this internship.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

## What I Can and Can't Claim

This project can identify observable patterns in anonymized search and content data to help prioritize which pages may deserve human review. The recommendations are intended to support human decision-making by ranking potential review candidates based on observable evidence from the available data.

However, this project cannot prove that refreshing a page will improve its performance or that any observed relationship represents a Google ranking factor. The results are observational and should be interpreted as decision-support rather than causal proof or guaranteed outcomes.

In [ ]:
observable_columns = [
    "impressions_90d",
    "clicks_90d",
    "sessions_90d",
    "content_age_days",
    "avg_position",
    "ctr"
]

print("Observable fields used in this project:")
for col in observable_columns:
    print(f"- {col}")

Observable fields used in this project:
- impressions_90d
- clicks_90d
- sessions_90d
- content_age_days
- avg_position
- ctr


## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.